bpf/cpu/cpu.bpf.c

```diff
@@ -747,9 +747,18 @@ int walk_user_stacktrace_impl(struct bpf_perf_event_data *ctx) {
         return 1;
       }
 
-      // TODO(sylfrena): add comments to explain calculations
+      // Stacktraces are saved return addresses from function calls pushed onto a stack
+      // The base pointer(bp) is first register pushed onto the stack and holds the beginning of the stack
+      // Then, the stack pointer points at the first frame at the bp updating the top of the stack to 8 bits ahead of the bp
+      // When the current instruction is pushed, top of the stack moves up by 1 frame, updating sp by another 8 bits
+      // Hence, we update current stack pointer by 2 bytes(16 bits) ahead of bp
       unwind_state->sp = unwind_state->bp + 16;
       unwind_state->bp = next_fp;
+      // Rewinding the program counter to get the instruction pointer for the previous function
+      // would be ideal but is unreliable in x86 due to variable width encoding.
```

Comment:
nit: Perhaps we should remark why. It is unreliable without disassembling the whole .text section, which is not practical